rsocket/internal/SetupResumeAcceptor.cpp

```diff
@@ -271,12 +279,11 @@ void SetupResumeAcceptor::closeAll() {
 
   closed_ = true;
 
-  std::runtime_error exn{"SetupResumeAcceptor is shutting down"};
-
   auto connections = std::move(connections_);
   for (auto& connection : connections) {
-    connection->onSubscribe(yarpl::flowable::Subscription::empty());
-    connection->onError(exn);
+    // Nasty, but we can't type the set as holding OneFrameSubscriber objects
+    // without exposing the OneFrameSubscriber definition in the header.
+    static_cast<OneFrameSubscriber*>(connection.get())->close();
```

Comment:
Expose it as `detail::OneFrameSubscriber` so we don't need to do any nasty casting, but make it clear this is a private implementation detail? This will eventually break when the type of `connections` changes at some point.

Reply:
Can't make it `detail::OneFrameSubscriber`.  There's a cyclical dependency between `SetupResumeAcceptor` and `OneFrameSubscriber` because `OFS` uses `SRA::OnSetup` and `SRA::OnResume`.

I'll look into making it a nested class instead.